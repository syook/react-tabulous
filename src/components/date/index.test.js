import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import DateComponent from '.';

configure({ adapter: new Adapter() });

const date = new Date();

describe('Date Component', () => {
  it('should render without crashing', () => {
    shallow(<DateComponent />);
  });

  it('matches snapshot', () => {
    const tree = shallow(<DateComponent />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should accept props', () => {
    const fn = jest.fn();
    const wrapper = mount(<DateComponent value={date} onChange={fn} />);
    expect(wrapper.props().value).toEqual(date);
    expect(wrapper.props().onChange).toEqual(fn);
  });

  it('props should accept proper datatypes', () => {
    const fn = jest.fn();
    const wrapper = mount(<DateComponent value={date} onChange={fn} />);
    expect(typeof wrapper.props().value).toEqual('object');
    expect(typeof wrapper.props().onChange).toEqual('function');
  });

  it('should render Date Picker', () => {
    const wrapper = mount(<DateComponent value={date} />);
    expect(wrapper.find('#date-picker').exists()).toBe(true);
  });

  it('Date Picker should accept selected prop', () => {
    const wrapper = shallow(<DateComponent value={date} />);
    expect(wrapper.find('#date-picker').prop('selected')).toBe(date);
  });

  it('Date Picker should accept date format prop', () => {
    const wrapper = shallow(<DateComponent value={date} />);
    expect(wrapper.find('#date-picker').prop('dateFormat')).toBe('MMMM d, yyyy');
  });
});
