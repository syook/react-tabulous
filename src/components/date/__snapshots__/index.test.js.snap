// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Component matches snapshot 1`] = `
<t
  allowSameDay={false}
  dateFormat="MMMM d, yyyy"
  dateFormatCalendar="LLLL yyyy"
  disabled={false}
  disabledKeyboardNavigation={false}
  dropdownMode="scroll"
  id="date-picker"
  inlineFocusSelectedMonth={false}
  monthsShown={1}
  nextMonthButtonLabel="Next Month"
  nextYearButtonLabel="Next Year"
  onBlur={[Function]}
  onCalendarClose={[Function]}
  onCalendarOpen={[Function]}
  onChange={[Function]}
  onClickOutside={[Function]}
  onFocus={[Function]}
  onInputClick={[Function]}
  onInputError={[Function]}
  onKeyDown={[Function]}
  onMonthChange={[Function]}
  onSelect={[Function]}
  onYearChange={[Function]}
  preventOpenOnFocus={false}
  previousMonthButtonLabel="Previous Month"
  previousYearButtonLabel="Previous Year"
  readOnly={false}
  renderDayContents={[Function]}
  shouldCloseOnSelect={true}
  showMonthYearPicker={false}
  showPopperArrow={true}
  showPreviousMonths={false}
  showQuarterYearPicker={false}
  showTimeInput={false}
  showTimeSelect={false}
  strictParsing={false}
  timeCaption="Time"
  timeInputLabel="Time"
  timeIntervals={30}
  withPortal={false}
/>
`;
